ding/example/ppo_with_complex_obs.py

```diff
@@ -0,0 +1,197 @@
+from typing import Union, Optional, List, Any, Tuple
+import os
+import torch
+import torch.nn as nn
+import numpy as np
+import gym
+from gym import spaces
+from ditk import logging
+from ding.envs import DingEnvWrapper, EvalEpisodeReturnEnv, \
+    BaseEnvManagerV2
+from ding.config import compile_config
+from ding.policy import PPOPolicy
+from ding.utils import set_pkg_seed
+from ding.model import VAC
+from ding.framework import task, ding_init
+from ding.framework.context import OnlineRLContext
+from ding.framework.middleware import multistep_trainer, StepCollector, interaction_evaluator, CkptSaver, \
+    gae_estimator, online_logger
+from easydict import EasyDict
+
+
+my_env_ppo_config = dict(
+    exp_name='my_env_ppo_seed0',
+    env=dict(
+        collector_env_num=4,
+        evaluator_env_num=4,
+        n_evaluator_episode=4,
+        stop_value=195,
+    ),
+    policy=dict(
+        cuda=True,
+        action_space='discrete',
+        model=dict(
+            obs_shape=None,
+            action_shape=2,
+            action_space='discrete',
+            critic_head_hidden_size=138,
+            actor_head_hidden_size=138,
+        ),
+        learn=dict(
+            epoch_per_collect=2,
+            batch_size=64,
+            learning_rate=0.001,
+            value_weight=0.5,
+            entropy_weight=0.01,
+            clip_ratio=0.2,
+            learner=dict(hook=dict(save_ckpt_after_iter=100)),
+        ),
+        collect=dict(
+            n_sample=256, unroll_len=1, discount_factor=0.9, gae_lambda=0.95, collector=dict(transform_obs=True, )
+        ),
+        eval=dict(evaluator=dict(eval_freq=100, ), ),
+    ),
+)
+my_env_ppo_config = EasyDict(my_env_ppo_config)
+main_config = my_env_ppo_config
+my_env_ppo_create_config = dict(
+    env_manager=dict(type='base'),
+    policy=dict(type='ppo'),
+)
+my_env_ppo_create_config = EasyDict(my_env_ppo_create_config)
+create_config = my_env_ppo_create_config
+
+
+class MyEnv(gym.Env):
+
+    def __init__(self, seq_len=5, feature_dim=10, image_size=(10, 10, 3)):
+        super().__init__()
+
+        # Define the action space
+        self.action_space = spaces.Discrete(2)
+
+        # Define the observation space
+        self.observation_space = spaces.Dict(
+            (
+                {
+                    '0': spaces.Dict(
```

Comment:
maybe you can use some more meaningful keys?

Reply:
fix